src/App.tsx

```diff
@@ -1,24 +1,134 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Completion } from './types/Completion';
+import { TodoError } from './types/TodoError';
+import { deleteTodoFromServer, getTodosFromServer } from './api/todos';
+import { USER_ID } from './constants';
+
+import { TodoForm } from './components/TodoForm';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { NotificationError } from './components/NotificationError';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(TodoError.NoError);
+  const [completionFilter, setCompletionFilter] = useState(Completion.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodosFromServer(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorMessage(TodoError.Load));
+  }, []);
+
+  const completedTodos = useMemo(() => (
```

Comment:
It's not a function, but a value obtained as a result of calculation. And memoization is used here since there's no point in calculating the value every render if conditions haven't changed.